app/bundles/EmailBundle/Mailer/Transport/AbstractTokenHttpTransport.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Mautic\EmailBundle\Mailer\Transport;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mailer\Exception\HttpTransportException;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/***
+ * This class is inspired of
+ * https://github.com/symfony/mailer/blob/6.1/Transport/AbstractHttpTransport.php
+ * The final implementation to send must inspire from
+ * https://github.com/symfony/mailer/blob/6.1/Transport/AbstractApiTransport.php
+ */
+
+abstract class AbstractTokenHttpTransport extends AbstractTokenArrayTransport implements TokenTransportInterface
+{
+    /**
+     * @var string|null
+     */
+    protected $host;
+
+    /**
+     * @var int|null
+     */
+    protected $port;
+
+    /**
+     * @var HttpClientInterface|null
+     */
+    protected $client;
+
+    /**
+     * @var string|null
+     */
+    private $username;
+
+    /**
+     * @var string|null
+     */
+    private $password;
+
+    /**
+     * @var string|null
+     */
+    private $apiKey;
```

Comment:
I don't like that we have a class that is trying to assume that some transport will use an API key and another will use username+password. There can be some other that will use another type of authentication. We'll have to figure out a way how the transports are free to configure their own authentication. I'll have to think more about this to find some suggestion that would work.

Reply:
`AbstractTokenHttpTransport.php` has been removed.